Examples/MAX78000/CNN/faceid_evkit-riscv/README.md

```diff
@@ -52,32 +54,85 @@ Connect USB cable to CN1 (USB/PWR) and turn ON power switch (SW1). Note the COM
 
 Connect PICO adapter to JH5 SWD header.
 
-If you are using Windows, load the firmware image with OpenOCD in a MinGW shell:
+If you are using Windows, open the MinGW window and start OpenOCD:
 
 ```bash
-openocd -s $MAXIM_PATH/Tools/OpenOCD/scripts -f interface/cmsis-dap.cfg -f target/max78000.cfg -c "program build/MAX78000.elf reset exit"
+openocd -s $MAXIM_PATH/Tools/OpenOCD/scripts -f interface/cmsis-dap.cfg -f target/max78000.cfg
 ```
 
-If using Linux, perform this step:
+Open a second MinGW window and load combined ARM/RISC-V application image using ARM GDB:
 
 ```bash
-./openocd -f tcl/interface/cmsis-dap.cfg -f tcl/target/max78000.cfg -c "program build/MAX78000.elf verify reset exit"
+arm-none-eabi-gdb build/max78000-combined.elf -x gdb.txt
 ```
 
-### Load firmware image to MAX78000 Feather
+If using Linux, perform these steps:
 
-Connect USB cable to CN1 USB connector.
+Start OpenOCD:
+
+```bash
+./openocd -f tcl/interface/cmsis-dap.cfg -f tcl/target/max78000.cfg
+```
+
+Load combined ARM/RISC-V application image using ARM GDB:
+
+```bash
+arm-none-eabi-gdb build/max78000-combined.elf -x gdb.txt
+```
+
+### Debugging application on MAX78000 EVKIT
+
+To debug the application, change optimization setting in project.mk:
+
+```bash
+# Set a higher optimization level to maximize performance
+#MXC_OPTIMIZE_CFLAGS = -O2
+# Default optimization level for debugging purpose
+MXC_OPTIMIZE_CFLAGS = -Og
+```
 
-If you are using Windows, load the firmware image with OpenOCD in a MinGW shell:
+disable the low power mode in **faceID.h **and recompile the code:
 
 ```bash
-openocd -s $MAXIM_PATH/Tools/OpenOCD/scripts -f interface/cmsis-dap.cfg -f target/max78000.cfg -c "program build/MAX78000.elf reset exit"
+//#define LP_MODE_ENABLE
 ```
 
-If using Linux, perform this step:
+Load and debug ARM/RISC-V application image using ARM GDB:
+
+```bash
+arm-none-eabi-gdb build/max78000-combined.elf
+(gdb) target remote localhost:3333
+(gdb) monitor reset halt
+(gdb) load
+(gdb) compare-sections
+(gdb) monitor reset halt
+(gdb) c
+```
+
+To debug an application on RISC-V, connect the Olimex adapter to JH4 and start OpenOCD:
+
+In Windows:
 
 ```bash
-./openocd -f tcl/interface/cmsis-dap.cfg -f tcl/target/max78000.cfg -c "program build/MAX78000.elf verify reset exit"
+openocd -s $MAXIM_PATH/Tools/OpenOCD/scripts -f interface/ftdi/olimex-arm-usb-ocd-h.cfg -f target/max78000_riscv.cfg
```

Comment:
RISC-V OpenOCD should be launched on a separate port

Ex: `openocd -c "gdb_port 3334" ...`